sway-ir/src/verify.rs

```diff
@@ -703,12 +713,16 @@ impl InstructionVerifier<'_, '_> {
         lhs_value: &Value,
         rhs_value: &Value,
     ) -> Result<(), IrError> {
-        // Comparisons must be between integers at this stage.
+        // Comparisons must be between integers or equivalent pointers at this stage.
         match (
             lhs_value.get_type(self.context),
             rhs_value.get_type(self.context),
         ) {
             (Some(lhs_ty), Some(rhs_ty)) => {
+                if lhs_ty.is_ptr(self.context) && rhs_ty.is_ptr(self.context) {
+                    // Not sure if it's okay to allow comparing pointers of different types.
```

Comment:
Without more context, I vote for no.

Reply:
Looks like the stricter check works fine.